File: backend/app/api/endpoints/results.py
"""
结果查询API端点
"""

from fastapi import APIRouter, HTTPException, Depends, Query
from sqlalchemy.orm import Session
from app.core.database import get_db
from typing import Dict, Any, List, Optional

router = APIRouter()


@router.get("/")
async def list_results(
    page: int = Query(1, ge=1, description="页码"),
    size: int = Query(20, ge=1, le=100, description="每页数量"),
    db: Session = Depends(get_db)
) -> Dict[str, Any]:
    """
    获取分析结果列表
    """
    try:
        # TODO: 从数据库查询结果列表
        # offset = (page - 1) * size
        # results = db.query(AnalysisResult).offset(offset).limit(size).all()
        # total = db.query(AnalysisResult).count()
        
        # 临时返回示例数据
        results = [
            {
                "id": "example-1",
                "filename": "drawing-001.pdf",
                "page_count": 3,
                "dimensions_found": 15,
                "created_at": "2024-12-01T10:00:00Z",
                "status": "completed"
            }
        ]
        
        return {
            "results": results,
            "pagination": {
                "page": page,
                "size": size,
                "total": len(results),
                "pages": 1
            }
        }
        
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"获取结果列表失败: {str(e)}"
        )


@router.get("/{result_id}")
async def get_result_detail(
    result_id: str,
    db: Session = Depends(get_db)
) -> Dict[str, Any]:
    """
    获取单个分析结果详情
    """
    try:
        # TODO: 从数据库查询具体结果
        # result = db.query(AnalysisResult).filter(AnalysisResult.id == result_id).first()
        # if not result:
        #     raise HTTPException(status_code=404, detail="结果不存在")
        
        # 临时返回示例数据
        result = {
            "id": result_id,
            "filename": "drawing-001.pdf",
            "file_size": 2048576,
            "page_count": 3,
            "dimensions": [
                {
                    "page": 1,
                    "value": "100.5",
                    "unit": "mm",
                    "tolerance": "±0.1",
                    "position": {"x": 150, "y": 200},
                    "confidence": 0.95
                },
                {
                    "page": 1,
                    "value": "50.0",
                    "unit": "mm",
                    "tolerance": None,
                    "position": {"x": 300, "y": 150},
                    "confidence": 0.88
                }
            ],
            "pages": [
                {
                    "page_number": 1,
                    "image_path": f"/images/{result_id}_page_1.png",
                    "dimensions_count": 2
                }
            ],
            "created_at": "2024-12-01T10:00:00Z",
            "processing_time": 45.2,
            "status": "completed"
        }
        
        return result
        
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"获取结果详情失败: {str(e)}"
        )


@router.get("/{result_id}/pages/{page_number}")
async def get_page_result(
    result_id: str,
    page_number: int,
    db: Session = Depends(get_db)
) -> Dict[str, Any]:
    """
    获取特定页面的分析结果
    """
    try:
        # TODO: 从数据库查询页面结果
        page_result = {
            "result_id": result_id,
            "page_number": page_number,
            "image_path": f"/images/{result_id}_page_{page_number}.png",
            "dimensions": [
                {
                    "value": "100.5",
                    "unit": "mm",
                    "tolerance": "±0.1",
                    "position": {"x": 150, "y": 200},
                    "confidence": 0.95
                }
            ]
        }
        
        return page_result
        
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"获取页面结果失败: {str(e)}"
        )


@router.delete("/{result_id}")
async def delete_result(
    result_id: str,
    db: Session = Depends(get_db)
) -> Dict[str, Any]:
    """
    删除分析结果
    """
    try:
        # TODO: 从数据库删除结果和相关文件
        return {
            "message": f"结果 {result_id} 已删除"
        }
        
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"删除结果失败: {str(e)}"
        )

File: frontend/src/api/results.ts
import api from './index'

export interface Dimension {
  page?: number
  value: string
  unit: string
  tolerance?: string
  position: { x: number; y: number }
  confidence: number
}

export interface PageResult {
  page_number: number
  image_path: string
  dimensions_count: number
}

export interface AnalysisResult {
  id: string
  filename: string
  file_size: number
  page_count: number
  dimensions: Dimension[]
  pages: PageResult[]
  created_at: string
  processing_time: number
  status: string
}

export interface ResultsListResponse {
  results: AnalysisResult[]
  pagination: {
    page: number
    size: number
    total: number
    pages: number
  }
}

export interface PageResultDetail {
  result_id: string
  page_number: number
  image_path: string
  dimensions: Dimension[]
}

/**
 * 获取分析结果列表
 */
export const getResultsList = async (page = 1, size = 20): Promise<ResultsListResponse> => {
  return api.get('/results/', {
    params: { page, size }
  })
}

/**
 * 获取单个分析结果详情
 */
export const getResultDetail = async (resultId: string): Promise<AnalysisResult> => {
  return api.get(`/results/${resultId}`)
}

/**
 * 获取特定页面的分析结果
 */
export const getPageResult = async (resultId: string, pageNumber: number): Promise<PageResultDetail> => {
  return api.get(`/results/${resultId}/pages/${pageNumber}`)
}

/**
 * 删除分析结果
 */
export const deleteResult = async (resultId: string): Promise<{ message: string }> => {
  return api.delete(`/results/${resultId}`)
}
